ComPact/Builders/V3/MessageBuilder.cs

```diff
@@ -75,6 +76,45 @@ public MessageBuilder VerifyConsumer<T>(Action<T> messageHandler)
             return this;
         }
 
+        /// <summary>
+        /// Invokes the provided message handler and checks that no exceptions are thrown.
+        /// </summary>
+        /// <typeparam name="T">Type to deserialize the defined message to.</typeparam>
+        /// <param name="messageHandler">Actual handling code of your message consumer.</param>
+        /// <returns></returns>
+        public async Task<MessageBuilder> VerifyConsumerAsync<T>(Func<T,Task> messageHandler)
+        {
+            CheckMessageHandler(messageHandler);
+
+            try
+            {
+                var serializedContent = JsonConvert.SerializeObject(_message.Contents);
+                var content = JsonConvert.DeserializeObject<T>(serializedContent);
```

Comment:
Since this is now duplicated, it may be extracted to a method.

Reply:
extracted to standalone code